src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -1219,6 +1233,28 @@ private HttpResponseHeaders CreateResponseHeaders(bool appCompleted)
         return responseHeaders;
     }
 
+    private bool CanIncludeResponseContentLengthHeader()
+    {
+        // Section 4.3.6 of RFC7231
+        if (Is1xxCode(StatusCode) || StatusCode == StatusCodes.Status204NoContent)
+        {
+            // A server MUST NOT send a Content-Length header field in any response
+            // with a status code of 1xx (Informational) or 204 (No Content).
+            return false;
+        }
+        else if (Method == HttpMethod.Connect && Is2xxCode(StatusCode))
```

Comment:
nit: single if with additional `|| (Method == HttpMethod.Connect && Is2xxCode(StatusCode))` ?

Reply:
Only if `Method == HttpMethod.Connect && Is2xxCode(StatusCode)` is moved to a local `Is2xxConnect()` local function or something. I don't like nested parens in conditions.

Reply:
Yea, I chose to do it this way because of readability (and it makes it clear which comments belong to each part). Will keep it as-is unless there's strong objection.